### ISSUE TITLE
Publish-Module Violates Single Responsibility Principle

### ISSUE THREAD

**Issue by fourpastmidnight** [PowerShell/PSResourceGet#1393]
### Summary of the new feature / enhancement

As a user, I expect Publish-Module to publish a module, not package _AND_ publish a module. As a user, I also expect that I can issue Package-Module to create a new module package that can then be published to one or more places, such as the PowerShellGallery and/or private package feeds.

### Proposed technical implementation details (optional)

For example, the following workflows are what I desire:

```powershell
PS C:\MyProject> Package-Module -Path ./out/MyModule

 ModuleName      Version    Path
 ----------      -------    -------------------------
 MyModule        1.0.0      C:\MyProject\artifacts\MyModule-1.0.0.nupkg

```

and then, combined with `Publish-Module`, perhaps something like:

```powershell
PS C:\MyProject> Package-Module -Path ./out/MyModule -PassThru | Publish-Module ...
```

**Comment by ThomasNieto** [comment]
I also have had the same concern for a while. I would have wanted a `New-PSResource` or `Build-PSResource` and publish only pushes the resource to the repository.

**Comment by ThomasNieto** [comment]
@SydneyhSmith I know this is close to release but the existing functionality is already there just splitting it into two cmdlets.

**Comment by SydneyhSmith** [comment]
We have been wanting to support publishing already built packages (nupkgs), and may need to in order to support certain repositories in the future...this won't block RC but I agree that it would be nice to have a `Build-PSResource` type command and support either unpacked modules or nupkgs as input for Publish-PSResource, probably as different parameter sets

**Comment by SydneyhSmith** [comment]
Today you can sort of accomplish this by using the `-DestinationPath` parameter on Publish

**Comment by leoniDEV** [comment]
The problem with the `-DestinationPath` is that the command still try to upload the package to remote repository and if you don't provide an API key the command fail (after blocking the command line for a while) with a 401 error

**Comment by JustinGrote** [comment]
I don't think it should be a separate cmdlet but `-DestinationPath` should be part of a separate parameterset. Unfortunately that would be abreaking change now for anyone already expecting it to do both.

As a workaround, you can `Register-PSResourceRepository -Name 'Local' -Uri /path/you/want/to/save/the/nupkg` and then publish to that and get the desired effect (make a nupkg without publishing to gallery/otherwise)
